### PR TITLE
datapath: remove fill-up notification for CT maps

### PR DESCRIPTION
**Pull request by julianwiedmann**
With https://github.com/cilium/cilium/pull/24378 merged, CT always uses LRU maps. So we no longer have to deal with fill-up of the CT maps.

### CODE REVIEW

File: pkg/signal/signal.go [hunk 1]
```diff
@@ -30,7 +30,7 @@ type SignalType uint32
 const (
 	// SignalNatFillUp denotes potential congestion on the NAT table
 	SignalNatFillUp SignalType = iota
-	// SignalCTFillUp denotes potential congestion on the CT table
+	// SignalCTFillUp denotes potential congestion on the CT table. Unused.
```
Comment (qmonnet) [inline pkg/signal/signal.go:33]: I'd maybe expand here, “Unused, given that the datapath no longer sends this signal”.

File: pkg/signal/signal_test.go [hunk 2]
```diff
@@ -77,16 +76,14 @@ func (t *signalSuite) TestSignalSet(c *C) {
...
 	c.Assert(sm.isSignalMuted(SignalAuthRequired), Equals, true)
 
 	// 2 active signals
-	err = sm.UnmuteSignals(SignalNatFillUp, SignalCTFillUp)
+	err = sm.UnmuteSignals(SignalNatFillUp, SignalAuthRequired)
 	c.Assert(err, IsNil)
 	c.Assert(sm.isMuted(), Equals, false)
 	c.Assert(sm.isSignalMuted(SignalNatFillUp), Equals, false)
-	c.Assert(sm.isSignalMuted(SignalCTFillUp), Equals, false)
-	c.Assert(sm.isSignalMuted(SignalAuthRequired), Equals, true)
```
Comment (qmonnet) [inline pkg/signal/signal_test.go:89]: Nit: We're losing this assert here, how about using another signal?

Same comment at a few other locations below

**Review by qmonnet — APPROVED** [review]
Looks good! Just a few, non-blocking nits below from my side

File: pkg/signal/signal_test.go (hunk 2 above)
Comment (julianwiedmann) [reply pkg/signal/signal_test.go:89]: There's no other signal left to test  :grin:. This is the best I could come up with on a quick look.

Comment (qmonnet) [reply pkg/signal/signal_test.go:89]: OK fair enough :)

**Review by ldelossa — APPROVED** [review]
LGTM

**Review by squeed — APPROVED** [review]

**Review by joestringer — CHANGES_REQUESTED** [review]
LRU map updates can get contended leading to this signal, we saw it in one production environment before. I can't share the details publicly but feel free to reach out to me for a private run-down.

I discussed this publicly a bit during LPC last year as well:
https://lpc.events/event/16/contributions/1368/

**Comment by joestringer** [comment]
Julian and I discussed this a bit out-of-band. My perspective on this is that the signal itself is high value with no noise when triggered with a LRU-backed connection tracking map. For example, under high load scenarios it triggers relatively rarely with LRU, but if you see this signal then it immediately indicates that there is very high contention. Users can react to this my looking for ways to better distribute traffic flows across their cluster.

At the same time, there are other considerations like how userspace reacts to this event. With a CT map backed by a hashtable, this may occur more frequently and reacting to this signal by triggering CT garbage collection likely makes sense. It may also trigger more often. On the other hand, for the LRU case, it's far less likely to trigger and when it does trigger it's likely that the load is very high on the system. Hence in that scenario it may not make sense to trigger GC any more.

**Comment by julianwiedmann** [comment]
> Julian and I discussed this a bit out-of-band. My perspective on this is that the signal itself is high value with no noise when triggered with a LRU-backed connection tracking map. For example, under high load scenarios it triggers relatively rarely with LRU, but if you see this signal then it immediately indicates that there is very high contention. Users can react to this metric my looking for ways to better distribute traffic flows across their cluster.
> 
> At the same time, there are other considerations like how userspace reacts to this event. With a CT map backed by a hashtable, this may occur more frequently and reacting to this signal by triggering CT garbage collection likely makes sense. On the other hand, for the LRU case, it's far less likely to trigger and when it does trigger it's likely that the load is very high on the system. Hence in that scenario it may not make sense to trigger GC any more.

thanks Joe for summarizing! Marking as draft for now - seems like we want to keep the signal around, but maybe still adjust whether it should (unconditionally) trigger GC.

**Pull request closed without merging**